tests/engines/registries/acceptance/draft/draft-test.ts

```diff
@@ -401,22 +401,20 @@ module('Registries | Acceptance | draft form', hooks => {
     test('metadata: contributor can remove herself', async assert => {
         const currentUser = server.create('user', 'loggedIn');
         const registrationSchema = server.schema.registrationSchemas.find('testSchema');
-        const branchedFrom = server.create('node');
+        const draftRegistration = server.create('draft-registration',
+            { registrationSchema, initiator: currentUser });
         const thisContributor = server.create('contributor', {
             users: currentUser,
             index: 0,
-            node: branchedFrom,
+            draftRegistration,
             permission: Permission.Write,
         });
-        server.createList('contributor', 1, { node: branchedFrom });
-        const draftRegistration = server.create('draft-registration',
-            { registrationSchema, initiator: currentUser, branchedFrom });
+        server.createList('contributor', 1, { draftRegistration });
 
         await visit(`/registries/drafts/${draftRegistration.id}/metadata`);
-
-        assert.dom(`[data-test-contributor-remove-self="${thisContributor.id}"] > button`)
+        assert.dom(`[data-test-contributor-remove="${thisContributor.id}"] > button`)
             .isVisible('remove me button is visible');
-        await click(`[data-test-contributor-remove-self="${thisContributor.id}"] > button`);
+        await click(`[data-test-contributor-remove="${thisContributor.id}"] > button`);
```

Comment:
Did I miss where this data-test-selector was changed in this PR, or is this test no longer testing the "remove self" functionality for write contributors?

Reply:
Remove self is a functionality on the `read-only` contributor cards. Since we have switched to using the editable user cards, every contributor, including the user itself, can be removed by clicking the "x" button.

Reply:
Yes, but non-admin (i.e. read-write) users should only get the read-only versions of these cards, so these tests should be for write contributors rather than admin contributors. They are explicitly for testing that functionality.

Reply:
I see. So what we need here is to render a `read-only` contributor widget if the user is not an admin. Otherwise, we render an editable widget. What seems to be lacking in the current API is a `currentUserPermission` field on `draft-registration`. I can add that to mirage and make use of that field here. The question is whether there is a PLAT ticket to do this.

Reply:
That's being [worked on now](https://openscience.atlassian.net/browse/ENG-2597).